### PR TITLE
[DATA-1345]: Add base command for common flags, add create-stream-key and render-image, use latest api-client

### PR DESCRIPTION
**Pull request by therockstorm**
## Summary

Requires https://github.com/Vertexvis/vertex-api-client-ts/pull/6.

Also fixes a partial file upload issue, skips already created parts, and allows you to pass a root component into `create-template` so you can generate one for a sub-assembly.

## Possible Regressions

n/a

## Dependencies

n/a

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
     "outDir": "lib",
     "rootDir": "src",
     "strict": true,
-    "target": "es2017"
+    "target": "es2020",
+    "lib": ["es2020"]
```
Comment (therockstorm) [inline tsconfig.json:11]: Allows me to use `Promise.allSettled`

File: src/commands/render-image.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  };
+
+  public async run(): Promise<void> {
+    const { args, flags } = this.parse(CreateTemplate);
+
+    const renderArgs = {
+      client: await VertexClient.build({
+        environment: flags.environment as Environment,
+      }),
+      renderReq: { id: args.id, height: 1000, width: 1000 },
```
Comment (danschultz) [inline src/commands/render-image.ts:43]: Can we make these sizes arguments?

**Review by danschultz — APPROVED** [review]
Small comment, otherwise LGTM.

File: src/commands/render-image.ts (hunk 2 above)
Comment (therockstorm) [reply src/commands/render-image.ts:43]: I meant to do that! Updated.

**Review by danschultz — APPROVED** [review]

**Pull request merged**